main/webapp/modules/core/scripts/facets/facet.js

```diff
@@ -35,6 +35,13 @@ class Facet {
   constructor(div, config, options) {
   	this._div = div;
   	this._config = config;
+  	
+  	if(this._config.nameAtCreation == "" || this._config.nameAtCreation == null)
+  	  this._config.nameAtCreation = config.name;
+  	  
+  	if(this._config.source == "" || this._config.source == null)
+  	  this._config.source = config.columnName;
+  	

```

Comment:
I am a bit confused by these new fields - I am not sure why it is necessary to introduce them? If we add them, doesn't that mean we should also add support for them in the backend (for instance so that those configuration fields appear in the JSON version of the history)?

Reply:
@wetneb : if the user modifies the name of the facet, and then wants to get back the original name, we must save it. The source is an indicator to help the user to know what this facet is about. 

I know this add quite some characters to the permalink, and that there is a limit on the URL length, but still, I believe it's worth.

As far as I know, nothing get sent from facets to the Java server.

Regards,
   Antoine

Reply:
I am not sure why it is useful to let the user restore the original name to be honest. I'd just keep it simple and only store the current name. For instance, for projects, we don't store the original name either, and it works just fine.

Reply:
@wetneb : I did that (add the name in the JSON of facets) because it's the way that the permalink will get it. It I want to retain both the changed name and the original name, I had to do this (add it to the JSON).

But, I will change that to just a ID, numerical, and I will be able to restore both new info I added. It will be clean and short.

Regards,
   Antoine

Reply:
Thanks! I think we should be able to implement this without adding any new field to the JSON representation of facets.